tvl2018.py

```diff
@@ -0,0 +1,1269 @@
+    """
+    aA = 0.01
+    aR = 0.00133
+
+    Nlt = np.zeros(len(Nst))
+
+    Nlt[0] = agc_next_frame(0, Nst[0], aA, aR)
+    for i in range(1, len(Nst)):
+        Nlt[i] = agc_next_frame(Nlt[i-1], Nst[i], aA, aR)
+
+    return Nlt
+
+
+def sound_field_to_cochlea(s: np.ndarray, filename_filter: str) -> np.ndarray:
+    """
+    Applies a filter to the sound field data to simulate cochlear processing.
+
+    Args:
+       s: The input sound field data, a 2D array with two columns for stereo signals.
+       filename_filter: The filename of the filter coefficients (MAT-file).
+
+    Returns:
+       out: The filtered output signal.
+    """
+
+    # Load the filter coefficients from the MAT file
+    filter_data = loadmat(f'{filename_filter}')
+    # Assuming vecCoefficients is a 1D array
+    vec_coefficients = filter_data['vecCoefficients'].flatten()
+
+    # Apply the filter to the left channel
+    out_left = convolve(s[:, 0], vec_coefficients, mode='full')
+
+    # Check if there's a right channel and apply the filter
+    if s.shape[1] > 1:
+        out_right = convolve(s[:, 1], vec_coefficients, mode='full')
+    else:
+        out_right = out_left
+
+    # Use vstack to maintain stereo format
+    out = np.vstack((out_left[1024:-1024], out_right[1024:-1024])).T
+    # Without filter, included in original code:
+    # out = np.vstack((np.zeros((1024, 2)), s, np.zeros((1024, 2))))
+    return out
+
+
+def synthesize_sound(frequency: float, duration: float, rate: int) -> np.ndarray:
+    """
+    Synthesizes a stereo sine wave sound with a given frequency.
+    Scaled to a level 10 dB below full scale, as done in original MATLAB test
+
+    Args:
+        frequency: Frequency of the sine wave in Hz.
+        duration: Duration of the sound in seconds.
+        rate: Sampling frequency in Hz.
+
+    Returns:
+        stero_sound: A 2D numpy array
+        representing the synthesized stereo sine wave.
+    """
+    # time array
+    t = np.linspace(0, duration, int(rate * duration), endpoint=False)
+    # Generate a full-scale sine wave (peak amplitude 1)
+    sine_wave = np.sin(2 * np.pi * frequency * t)
+    scale = 10 ** ((-10) / 20)
+    scaled_sine_wave = scale * sine_wave
+
+    stereo_sound = np.column_stack((scaled_sine_wave, scaled_sine_wave))
+
+    return stereo_sound
+
+
+def main_tv2018(filename_or_sound: Union[str, np.ndarray],
+                db_max: float, filename_filter: str,
+                output_path: str = None, rate: int = None):
+    """
+    Calculate loudness according to Moore, Glasberg & Schlittenlacher (2016).
+
+    Args:
+        filename_or_sound: The path to filename of the sound file, the audio data
+                itself, or synthesized sound frequency (ex. synthesize_1000khz100ms)
+        db_max: The root-mean-square sound pressure level of a full-scale sinusoid,
+                i.e. a sinusoid whose peak amplitude is 1 in Matlab.
+
+        filename_filter: The filename of the FIR filter.
```

Comment:
Ick again.  Reading files from python is a pain, since the .mat file is supplied by you, but it won't necessarily be installed where it is easy for a user.